src/main/java/io/jenkins/plugins/gitlabbranchsource/GitLabHookCreator.java

```diff
@@ -160,24 +159,6 @@ public static String getHookUrl(boolean isWebHook) {
             .expand();
     }
 
-    static void checkURL(String url) {
-        try {
-            URL anURL = new URL(url);
-            if ("localhost".equals(anURL.getHost())) {
-                throw new IllegalStateException(
-                    "Jenkins URL cannot start with http://localhost \nURL is: " + url);
-            }
-            if (!anURL.getHost().contains(".")) {
```

Comment:
@baymac The GitLab might be in a private environment. So `localhost` might be the right address.

Reply:
An alternative solution could provide an administrative monitor if Jenkins URL does not meet some conditions.